### PR TITLE
[eclipse/xtext#1629] ported more code from Xtend to Java

### PR DESCRIPTION
**Pull request by cdietrich**
[eclipse/xtext#1629] ported more code from Xtend to Java
Signed-off-by: Christian Dietrich <christian.dietrich@itemis.de>

### CODE REVIEW

File: org.eclipse.xtext.java/src/org/eclipse/xtext/java/resource/ClassFileCache.java [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+		@Override
+		public boolean isAdapterForType(Object object) {
+			return object == ClassFileCache.class;
+		}
+	}
+
+	private static final Object NULL = new Object();
+
+	private final Map<QualifiedName, Object> cache = new ConcurrentHashMap<QualifiedName, Object>();
```
Comment (szarnekow) [inline org.eclipse.xtext.java/src/org/eclipse/xtext/java/resource/ClassFileCache.java]: A few unnecessary type arguments. The diamond op could help to simplify the code.

File: org.eclipse.xtext.java/src/org/eclipse/xtext/java/resource/ClassFileCache.java [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+		return false;
+	}
+
+	public void addCompileResult(char[] fileName, List<String> topLevelTypes, Map<String, byte[]> classMap) {
+		allTopLevelTypes.put(fileName, topLevelTypes);
+		allClassMaps.put(fileName, classMap);
+	}
+
+	public static ClassFileCache findInEmfObject(Notifier emfObject) {
+		for (Adapter adapter : emfObject.eAdapters()) {
```
Comment (szarnekow) [inline org.eclipse.xtext.java/src/org/eclipse/xtext/java/resource/ClassFileCache.java]: Should use EcoreUtil.getExistingAdapter to benefit from the new EScannableAdapterList

File: org.eclipse.xtext.java/src/org/eclipse/xtext/java/resource/ClassFileCache.java [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+			if (adapter instanceof ClassFileCache.ClassFileCacheAdapter) {
+				return ((ClassFileCache.ClassFileCacheAdapter) adapter).get();
+			}
+		}
+		return null;
+	}
+
+	public static ClassFileCache removeFromEmfObject(Notifier emfObject) {
```
Comment (szarnekow) [inline org.eclipse.xtext.java/src/org/eclipse/xtext/java/resource/ClassFileCache.java]: Dead code?

File: org.eclipse.xtext.java/src/org/eclipse/xtext/java/resource/JavaConfig.java [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+	public static JavaConfig findInEmfObject(Notifier emfObject) {
+		for (Adapter adapter : emfObject.eAdapters()) {
+			if (adapter instanceof JavaConfig.JavaConfigAdapter) {
+				return ((JavaConfig.JavaConfigAdapter) adapter).get();
+			}
+		}
+		return null;
+	}
+
+	public static JavaConfig removeFromEmfObject(Notifier emfObject) {
```
Comment (szarnekow) [inline org.eclipse.xtext.java/src/org/eclipse/xtext/java/resource/JavaConfig.java]: Dead code?

File: org.eclipse.xtext.java/src/org/eclipse/xtext/java/resource/JavaConfig.java [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+			return object == JavaConfig.class;
+		}
+	}
+
+	private JavaVersion javaSourceLevel;
+
+	private JavaVersion javaTargetLevel;
+
+	public static JavaConfig findInEmfObject(Notifier emfObject) {
+		for (Adapter adapter : emfObject.eAdapters()) {
```
Comment (szarnekow) [inline org.eclipse.xtext.java/src/org/eclipse/xtext/java/resource/JavaConfig.java]: EcoreUtil.getExistingAdapter

File: org.eclipse.xtext.java/src/org/eclipse/xtext/java/resource/JavaResourceServiceProvider.java [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+public class JavaResourceServiceProvider implements IResourceServiceProvider {
+	@Inject
+	private Injector injector;
+
+	@Inject
+	private FileExtensionProvider extensionProvider;
+
+	@Override
+	public boolean canHandle(final URI uri) {
+		return (extensionProvider.getFileExtensions().contains(uri.fileExtension()) && !uri.isArchive());
```
Comment (szarnekow) [inline org.eclipse.xtext.java/src/org/eclipse/xtext/java/resource/JavaResourceServiceProvider.java]: unnecessary parentheses

File: org.eclipse.xtext.java/src/org/eclipse/xtext/java/resource/ClassFileCache.java (hunk 3 above)
Comment (cdietrich) [reply org.eclipse.xtext.java/src/org/eclipse/xtext/java/resource/ClassFileCache.java]: am not sure if we should keep if somebody uses it

Comment (szarnekow) [reply org.eclipse.xtext.java/src/org/eclipse/xtext/java/resource/ClassFileCache.java]: Agreed

File: org.eclipse.xtext.java/src/org/eclipse/xtext/java/resource/JavaConfig.java [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+			return object == JavaConfig.class;
+		}
+	}
+
+	private JavaVersion javaSourceLevel;
+
+	private JavaVersion javaTargetLevel;
+
+	public static JavaConfig findInEmfObject(Notifier emfObject) {
+		return (JavaConfig) EcoreUtil.getExistingAdapter(emfObject, JavaConfig.class);
```
Comment (szarnekow) [inline org.eclipse.xtext.java/src/org/eclipse/xtext/java/resource/JavaConfig.java]: Must be `((JavaConfigAdapter)getExistingAdapter(emfObject, JavaConfig.class)).get())`

**Pull request merged**